pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -410,3 +373,123 @@ func executeAgentCommands(commands []AgentCommand, con cli.Connection, agentStor
 		}
 	}
 }
+
+func fetchAndStoreGuestInfo(commands []AgentCommand, con cli.Connection, agentStore *AsyncAgentStore, domainName string) {
+	domain, err := con.LookupDomainByName(domainName)
+	if err != nil {
+		// Skip execution on error, domain lookup failed
+		return
+	}
+	defer domain.Free()
+
+	flags := libvirt.DOMAIN_GUEST_INFO_INTERFACES |
```

Comment:
Why do we have different polling intervals (L290-L314), when we query the complete set of information every time?